components/job-orchestration/job_orchestration/scheduler/query/query_scheduler.py

```diff
@@ -821,12 +947,18 @@ async def handle_finished_extract_ir_job(
         duration=(datetime.datetime.now() - job.start_time).total_seconds(),
     ):
         if new_job_status == QueryJobStatus.SUCCEEDED:
-            logger.info(f"Completed IR extraction job {job_id}.")
+            logger.info(f"Completed stream extraction job {job_id}.")
         else:
-            logger.info(f"Completed IR extraction job {job_id} with failing tasks.")
+            logger.info(f"Completed stream extraction job {job_id} with failing tasks.")
+
+    waiting_jobs: List[str]
+    if QueryJobType.EXTRACT_IR == job.get_type():
+        extract_ir_config: ExtractIrJobConfig = job.get_config()
+        waiting_jobs = active_file_split_ir_extractions.pop(extract_ir_config.file_split_id)
+    else:
+        extract_json_config: ExtractJsonJobConfig = job.get_config()
+        waiting_jobs = active_archive_json_extractions.pop(extract_json_config.archive_id)
 
-    waiting_jobs = active_file_split_ir_extractions[file_split_id]
-    waiting_jobs.remove(job_id)
     for waiting_job in waiting_jobs:
```

Comment:
Since we don't remove the job that actually did the extraction first, aren't we going to attempt to set that job's status twice?

Reply:
Shouldn't have removed this line. thanks for catching